### DISCUSSION TITLE
Is it possible to get an opengl version of videoStream_vert.glsl vertex shader?

### DISCUSSION THREAD

**Discussion by Jia-WeiFang** [korejan/ALVR#69] [Q&A] [answered]
Hi,

I want to do the unwarping work offline using an opengl project.

But there are quite a lot different things in the vertex shader between the ALVR opengl version **QUAD_2D_VERTEX_SHADER** and the Vulkan version **videoStream_vert.glsl**.

May I ask how to convert the Vulkan version into an opengl one?

Thanks a lot for your kind help.

**Comment by korejan** [comment, marked as answer]
Hi, the alxr render code is quit different from the alvr client. Currently there is only partial implementation for opengl(es) in alxr, the vertex shader is simple and extensions used is support by opengl. You should only need to change very little or use opengl extensions to support loading spir-v bytecode.
